R/format_PFN.R

```diff
@@ -0,0 +1,589 @@
+
+  }
+
+}
+
+
+
+#' Create brood data table for EastDartmoor.
+#'
+#' @param Primary_data Primary data from EastDartmoor.
+#' @param ReRingTable Table containing ring numbers and unique identifiers for re-ringed individuals
+#'
+#' @return A data frame with Brood data
+
+
+create_brood_EDM <- function(Primary_data, ReRingTable){
+
+  ## Pre) Determine a vector of "bad" (nonconclusive) IDs
+  #This will be used downstream in point 7)
+  badIDs <- c("RUNT", "ringed", "ringed left", "ringed right", "unringed", "Unringed")
+
+
+  ## 1) Rename columns that are equivalent (content and format) to columns in the standard format
+  Brood_data <- Primary_data %>%
+    dplyr::rename(Plot = .data$Popn,
+                  LocationID = .data$Box) %>%
+
+    ## 2) Add population ID and reformat colums with equivalent content but different format
+    dplyr::mutate(PopID = "EDM",
+                  BreedingSeason = as.integer(Year),
+                  Species = dplyr::case_when(.data$Species == "BLUTI" ~ Species_codes[Species_codes$SpeciesID == 14620, ]$Code,
+                                             .data$Species == "PIEFL" ~ Species_codes[Species_codes$SpeciesID == 13490, ]$Code,
+                                             .data$Species == "GRETI" ~ Species_codes[Species_codes$SpeciesID == 14640, ]$Code,
+                                             .data$Species == "REDST" ~ Species_codes[Species_codes$SpeciesID == 11220, ]$Code,
+                                             .data$Species == "MARTI" ~ Species_codes[Species_codes$SpeciesID == 14400, ]$Code,
+                                             .data$Species == "NUTHA" ~ Species_codes[Species_codes$SpeciesID == 14790, ]$Code,
+                                             .data$Species == "COATI" ~ Species_codes[Species_codes$SpeciesID == 14610, ]$Code,
+                                             .data$Species == "WREN" ~ NA_character_, # Missing, 1 observation only
+                                             .data$Species == "TREEC" ~ NA_character_), # Missing, 1 observation only
+                  ClutchType = dplyr::case_when(.data$CltCd == "1" ~ "first",
+                                                .data$CltCd == "2" ~ "replacement",
+                                                .data$CltCd == "3" ~ "second"),
+                  LayDate = as.Date(paste('31/03/', .data$BreedingSeason, sep = ''), format = "%d/%m/%Y") + as.numeric(.data$DFE),
+                  ClutchSize = dplyr::case_when(as.integer(.data$CltSize) > 0 ~ as.integer(.data$CltSize),
+                                                as.integer(.data$CltSize) == 0 | is.na(.data$CltSize) ~ NA_integer_),
+                  ClutchSize_min = dplyr::case_when(is.na(.data$CltSize) | as.integer(.data$CltSize) == 0 ~ suppressWarnings(pmax(as.integer(.data$MinEggs), as.integer(.data$UnHatch)+as.integer(.data$Hatch), as.integer(.data$UnHatch)+as.integer(.data$Fledged), na.rm = TRUE))),
```

Comment:
I've added a comment below this now to explain why we use the maximum to determine the minimum. I hope it makes sense: 

_"In cases in which clutch size was either not recorded, or recorded as 0 (e.g. when clutches were laid but not incubated), other types of data may contain information on the minimum size of the clutch:
 Minimum number of eggs laid (MinEggs), and number of unhatched eggs (Unhatch) plus number of hatched eggs (Hatch) or fledglings (Fledged). Usually, only some of this information is available, and we therefore define the minimum possible clutch size as the largest number in the above categories. We use the largest number to determine minimum clutch size because all counted eggs/chicks were definitely part of the clutch, but the chances of eggs/chicks having disappeared before the count increases as we move further away from the laying date (i.e. number of fledlings (+ number of unhatched eggs) <= number of hatched eggs (+ number of unhatched eggs) <= number of laid eggs)"_

Reply:
If you think this is inappropriate / confusing, we can remove this and just provide ClutchSize_observed = NA with no minimum number when clutch size was not explicitly recorded in the raw data.

Reply:
It is a bit confusing, but that's just the nature of the data so I think it's fine to keep it in there with an informative comment 😃